stko/molecular/topology_extractor/topology_extractor.py

```diff
@@ -0,0 +1,73 @@
+"""
+Topology Extractor
+==================
+
+Class for defining a topology from a molecule and disconnections.
+
+"""
+
+import logging
+
+from ..networkx import Network
+
+logger = logging.getLogger(__name__)
+
+
+class TopologyExtractor:
+    """
+    Extractor of topology definitions from a molecule.
+
+    """
+
+    def extract_topology(
+        self,
+        molecule,
+        atom_ids_to_disconnect,
+    ):
+
+        flat_tuple = []
+        for i in atom_ids_to_disconnect:
+            flat_tuple.append(i[0])
+            flat_tuple.append(i[1])
+        flat_tuple = tuple(set(flat_tuple))
+
+        connected_graphs = self.get_connected_graphs(
+            molecule=molecule,
+            atom_ids_to_disconnect=atom_ids_to_disconnect,
+        )
+
+        # Get centroids.
+        self._centroids = {}
```

Comment:
This is a bit weird because if someone calls `get_vertex_positions` before they call `extract_topology`, they will get an error. I would just make `extract_topology` return a class like

```
class TopologyData:
    def __init__(self, centroids, connectivities):
        self.centroids = centroids
        self.connectivities = connectivities
```

and remove the `get_vertex_positions` and `get_connectivities` methods.

Reply:
Agreed and implemented.